language/functional-tests/src/evaluator.rs

```diff
@@ -343,7 +343,10 @@ fn run_transaction(
                     Err(ErrorKind::VMExecutionFailure(output).into())
                 }
             }
-            TransactionStatus::Discard(_) => Err(ErrorKind::DiscardedTransaction(output).into()),
+            TransactionStatus::Discard(_) => {
+                assert!(output.write_set().is_empty());
```

Comment:
I would certainly appreciate it if you could rather use checked_verify! from mirai-annotations v1.6.0. It generates the exact same code as present when MIRAI not the compiler, but gives MIRAI more information than assert does because it distinguishes the path condition from the verification condition.